File: app/src/main/java/com/example/katakonst/licenta/ItemDetailFragment.java
package com.example.katakonst.licenta;

import android.app.Activity;
import android.media.AudioManager;
import android.media.MediaPlayer;
import android.support.design.widget.CollapsingToolbarLayout;
import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.example.katakonst.licenta.JsonModels.Tracks;
import com.example.katakonst.licenta.dummy.DummyContent;

import org.apache.commons.codec.net.URLCodec;

/**
 * A fragment representing a single Item detail screen.
 * This fragment is either contained in a {@link ItemListActivity}
 * in two-pane mode (on tablets) or a {@link ItemDetailActivity}
 * on handsets.
 */
public class ItemDetailFragment extends Fragment {
    /**
     * The fragment argument representing the item ID that this fragment
     * represents.
     */
    public static final String ARG_ITEM_ID = "item_id";

    private  String nume;
    /**
     * The dummy content this fragment is presenting.
     */
    private Tracks mItem;

    /**
     * Mandatory empty constructor for the fragment manager to instantiate the
     * fragment (e.g. upon screen orientation changes).
     */
    public ItemDetailFragment() {
    }

    public String getNume()
    {
        return  nume;
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        if (getArguments().containsKey(ARG_ITEM_ID)) {
             mItem = (Tracks)getArguments().get("track");
             nume=getArguments().getString("nume");
           if(nume==null) {
               Bundle args = getActivity().getIntent().getExtras();
               nume = args.getString("nume");
           }
Log.d("s",nume);
            Activity activity = this.getActivity();
            CollapsingToolbarLayout appBarLayout = (CollapsingToolbarLayout) activity.findViewById(R.id.toolbar_layout);
            if (appBarLayout != null) {
                appBarLayout.setTitle(nume);
            }

            try {
                String url = "http://192.168.1.24:2000?test="+new URLCodec().encode(nume);
                Log.d("S",url);
                MediaPlayer mPlayer = new MediaPlayer();
                mPlayer.setAudioStreamType(AudioManager.STREAM_MUSIC);
                mPlayer.setDataSource(url);
                mPlayer.prepare();
                mPlayer.start();
              //  mPlayer.seekTo(789);
            }
            catch(Exception ex){
                Log.d("d",ex.toString());

            }
        }
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        View rootView = inflater.inflate(R.layout.item_detail, container, false);

        if (nume != null) {
            ((TextView) rootView.findViewById(R.id.item_detail)).setText(nume);
        }
        else {
Log.d("null","null");
        }

        return rootView;
    }
}

File: app/src/main/java/com/example/katakonst/licenta/ItemDetailFragment.kt
package com.example.katakonst.licenta

import android.content.Intent
import android.media.AudioManager
import android.media.MediaPlayer
import android.os.Bundle
import android.support.v4.app.Fragment
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.*
import com.example.katakonst.licenta.AsyncTask.Comments.ShowCommentsTask
import com.example.katakonst.licenta.AsyncTask.HashTags.GetHashTagsOfTrack
import com.example.katakonst.licenta.AsyncTask.ShowLikeNrTask
import com.example.katakonst.licenta.AsyncTask.users.DownloadImageTask
import com.example.katakonst.licenta.JsonModels.Comments
import com.example.katakonst.licenta.JsonModels.HashTags

import com.example.katakonst.licenta.JsonModels.Tracks

import org.apache.commons.codec.net.URLCodec
import java.util.*


class ItemDetailFragment : Fragment() {

    var nume: String? = null
    var id: String? = null
        private set
    var link: String? = null
    var photoLink:String?=null;
    private var tvw: View? = null
    private var mItem: Tracks? = null


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)


            val args = activity.intent.extras
            link = args.getString(Tracks.Companion.Link)
            nume = args.getString(Tracks.Companion.Name)
            id=args.getString(Tracks.Companion.Id).toString()
            photoLink=args.getString(Tracks.Companion.PhotoLink);
            Log.d("s", nume)
            val activity = this.activity

        try {
                val url = Constants.streamip + "/" + URLCodec().encode(link)
                Log.d("S", url)
                val mPlayer = MediaPlayer()
                mPlayer.setAudioStreamType(AudioManager.STREAM_MUSIC)
                mPlayer.setDataSource(url)
                mPlayer.prepare()
                val duration = mPlayer .duration
                val seekBar = this.activity.findViewById(R.id.seekBar3) as SeekBar
                val likeText = this.activity.findViewById(R.id.LikeNrText) as TextView
                val task = ShowLikeNrTask(likeText, id)
                task.execute()
                val playBut = this.activity.findViewById(R.id.button) as Button
                val likeButton = this.activity.findViewById(R.id.LikeButton) as Button
                val trackPhoto = this.activity.findViewById(R.id.trackPhoto) as ImageView
                val trackPhotoThread=DownloadImageTask(trackPhoto,Constants.streamip+"/"+photoLink)
                trackPhotoThread.execute();
               var hashTagList=this.activity.findViewById(R.id.TrackHashTags) as ListView

            var hashTagAdapter=ArrayAdapter<HashTags>(hashTagList.context,
                    android.R.layout.simple_list_item_1,
                    ArrayList<HashTags>());
            hashTagList.adapter=hashTagAdapter

            val hashTagTask = GetHashTagsOfTrack(hashTagAdapter, id)
            hashTagTask.execute()




            likeButton.setOnClickListener{
                var intent=Intent(this.context, UsersWhoLikedTrack::class.java);
                var bundle=Bundle()
                bundle.putString(Tracks.Id,id)
                intent.putExtras(bundle)
                this.activity.startActivity(intent)

             }

                playBut.setOnClickListener {

                    if (playBut.text == "Play") {
                        playBut.text = "Pause"

                        mPlayer.start()
                    } else {
                        mPlayer.pause()
                        playBut.text = "Play"

                    }
                }
                var list=this.activity.findViewById(R.id.listView3) as ListView

                var adaprter=ArrayAdapter<Comments>(list.context,
                        android.R.layout.simple_list_item_1,
                        ArrayList<Comments>());
                list.adapter=adaprter

                val commentsTask = ShowCommentsTask(adaprter, id)
                commentsTask.execute()



                seekBar.max = duration
                seekBar.setOnSeekBarChangeListener(object : SeekBar.OnSeekBarChangeListener {
                    internal var progress = 0

                    override fun onProgressChanged(seekBar: SeekBar, progresValue: Int, fromUser: Boolean) {
                        progress = progresValue
                        Log.d("progress", progress.toString())
                        mPlayer.seekTo(progress)

                    }

                    override fun onStartTrackingTouch(seekBar: SeekBar) {
                    }

                    override fun onStopTrackingTouch(seekBar: SeekBar) {
                    }
                })








                //  mPlayer.seekTo(789);
            } catch (ex: Exception) {
                Log.d("d", ex.toString())

            }

        }


    override fun onCreateView(inflater: LayoutInflater?, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        val rootView = inflater!!.inflate(R.layout.item_detail, container, false)
        tvw = rootView
        Log.d("bossss", tvw.toString())



        if (nume != null) {
            (rootView.findViewById(R.id.item_detail) as TextView).text = nume
        } else {
            Log.d("null", "null")
        }

        return rootView
    }

    companion object {
        /**
         * The fragment argument representing the item ID that this fragment
         * represents.
         */
        val ARG_ITEM_ID = "item_id"
    }
}
/**
 * Mandatory empty constructor for the fragment manager to instantiate the
 * fragment (e.g. upon screen orientation changes).
 */
